src/cogent3/core/new_alignment.py

```diff
@@ -424,7 +393,7 @@ def get_seq_bytes(
         return self.get_seq_str(seqid=seqid, start=start, stop=stop).encode("utf8")
 
     def get_view(self, seqid: str) -> SeqDataView:
-        seq_len = len(self._data[seqid])
+        seq_len = self.seq_lengths()[seqid]
```

Comment:
I would just directly access the attribute rather than the method. Also, I think we may be tying the API too closely to our implementation. How about we have a method `get_seq_length(name)` which users can get access to. This means the implementation does not have to create a dictionary and return it. Or worse yet, create a dictionary (which is mutable) and the user modifies it and gets unexplained behaviour 😳

Just add that as a design note on the method, don't implement yet